src/sdk/pynni/nni/tuner.py

```diff
@@ -93,3 +93,16 @@ def _on_exit(self):
 
     def _on_error(self):
         pass
+
+    def extract_scalar_reward(self, value, scalar_key='default'):
+        if isinstance(value, float) or isinstance(value, int):
```

Comment:
I suggest to differ 'return' value from 'param' value to eliminate confusion, such as:

if isinstance(value, float) or isinstance(value, int):
        resultValue = value
        elif isinstance(value, dict) and scalar_key in value:
            resultValue = value[scalar_key]
            if isinstance(resultValue, float) or isinstance(resultValue, int):
                pass
            else:
                raise RuntimeError('Incorrect final result: the final result for %s should be float/int, or a dict which has a key named "default" whose value is float/int.' % str(self.__class__))
        else:
            raise RuntimeError('Incorrect final result: the final result for %s should be float/int, or a dict which has a key named "default" whose value is float/int.' % str(self.__class__)) 
        return resultValue